### PR TITLE
Feature/server save

### PR DESCRIPTION
**Pull request by ZQUEMA**
This first pull request adds a refactoring of the database by adding fields, removing unnecessary functionality and the ability to store and load channel data to/from a file.

### CODE REVIEW

File: libs/database/tables/teams/team.c [hunk 1]
```diff
@@ -12,21 +12,18 @@
...
-        perror("Team creation");
+    if (!team)
         exit (84);
-    }
     team->id = id;
     team->uuid = uuid_gen();
     team->name = strdup(name);
-    if (!team->uuid || !team->name) {
-        perror("Team creation");
+    team->name = strdup(desc);
```
Comment (aurelenc) [inline libs/database/tables/teams/team.c:24]: name ?

**Review by aurelenc — CHANGES_REQUESTED** [review]

**Review by aurelenc — APPROVED** [review]

File: libs/database/tables/channels/database_channels_save.c [hunk 2]
```diff
@@ -32,7 +32,7 @@ bool db_save_channels(database_t *db, const char *filepath)
 
     fptr = fopen(filepath, "w");
```
Comment (aurelenc) [inline libs/database/tables/channels/database_channels_save.c:33]: w+ ?

File: libs/database/tables/load_functions.c [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        buffer = calloc(sizeof(char), length + 1);
+        if (buffer) {
+            fread (buffer, 1, length, f);
+        }
+        fclose (f);
+    }
+    return (buffer);
+}
+
+char *get_arg(char *entity, size_t *cur)
```
Comment (TomDUVAL-MAHE) [inline libs/database/tables/load_functions.c:32]: cur ???

Comment (ZQUEMA) [reply libs/database/tables/load_functions.c:32]: cursor (that's for the Epitech C Coding Style)

**Pull request merged**